src/drafty.js

```diff
@@ -2295,14 +2335,17 @@ function spansToTree(parent, text, start, end, spans) {
       type: span.type,
       data: span.data,
       key: span.key
-    }, text, start, span.end, subspans));
+    }, graphemes, start, span.end, subspans));
     start = span.end;
   }
 
   // Add the last unformatted range.
   if (start < end) {
     addNode(parent, {
-      text: text.substring(start, end)
+      text: graphemes
+        .slice(start, end)
+        .map((segment) => segment.segment)
+        .join("")
```

Comment:
Code style: please use ' (single quotes) for strings which require no localization.